tests/foreman/api/test_location.py

```diff
@@ -199,3 +200,276 @@ def test_create_location_with_multiple_organization(self):
+
+        @Feature: Location - Update
+
+        """
+        location = entities.Location(
+            subnet=[entities.Subnet().create()],
+        ).create()
+        subnet = entities.Subnet().create()
+        location.subnet = [subnet]
+        self.assertEqual(location.update(['subnet']).subnet[0].id, subnet.id)
+
+    def test_update_location_with_new_environment(self):
+        """@Test: Update location with new environment
+
+        @Assert: Location updated successfully and has correct environment
+        assigned
+
+        @Feature: Location - Update
+
+        """
+        location = entities.Location(
+            environment=[entities.Environment().create()],
+        ).create()
+        env = entities.Environment().create()
+        location.environment = [env]
+        self.assertEqual(
+            location.update(['environment']).environment[0].id,
+            env.id
+        )
+
+    def test_update_location_with_new_host_group(self):
+        """@Test: Update location with new host group
+
+        @Assert: Location updated successfully and has correct host group
+        assigned
+
+        @Feature: Location - Update
+
+        """
+        location = entities.Location(
+            hostgroup=[entities.HostGroup().create()],
+        ).create()
+        host_group = entities.HostGroup().create()
+        location.hostgroup = [host_group]
+        self.assertEqual(
+            location.update(['hostgroup']).hostgroup[0].id,
+            host_group.id
+        )
+
+    def test_update_location_with_new_organization(self):
+        """@Test: Update location with new organization
+
+        @Assert: Location updated successfully and has correct organization
+        assigned
+
+        @Feature: Location - Update
+
+        """
+        location = entities.Location(
+            organization=[entities.Organization().create()],
+        ).create()
+        org = entities.Organization().create()
+        location.organization = [org]
+        self.assertEqual(
+            location.update(['organization']).organization[0].id,
+            org.id
+        )
+
+    def test_update_location_with_multiple_organizations(self):
+        """@Test: Update location with with multiple organizations
+
+        @Assert: Location created successfully and has correct organizations
+        assigned
+
+        @Feature: Location - Update
+
+        """
+        location = entities.Location(
+            organization=[entities.Organization().create()],
+        ).create()
+        orgs_amount = randint(3, 5)
+        orgs = [
+            entities.Organization().create() for _ in range(orgs_amount)
+        ]
+        location.organization = orgs
+        location = location.update(['organization'])
+        self.assertEqual(len(location.organization), orgs_amount)
+        for organization in location.organization:
+            self.assertIn(organization.id, [org.id for org in orgs])
```

Comment:
unittest provides great error messages when an assertion is made against a set. For example, if I wrote this:

```python
self.assertEqual(set([1, 2, 3]), set([2, 3, 4]))
```

I would get an error message something like this:

    Items in the first set but not the second: [1]
    Items in the second set but not the first: [4]

With that in mind, here's an alternate form for this test:

```python
        location = entities.Location(
            organization=[entities.Organization().create()],
        ).create()
        orgs = [
            entities.Organization().create() for _ in range(randint(3, 5))
        ]
        location.organization = orgs
        location = location.update(['organization'])
        self.assertEqual(
            set([org.id for org in orgs]),
            set([org.id for org in location.organization]),
        )
```